### PR TITLE
Improve Multi-Instance Docker Support and DB Restoration Fix

### PR DESCRIPTION
**Pull request by gabepetersen**
This PR will accomplish two tasks:

1. The ability for multiple Optimizely Docker projects to be created without storage conflicts in /var/opt/mssql/host_data
    - This commit 7a2b9b56d48050f51e4f6f48b76f157aa233bfb1 will add a .env file to each Docker project that specifies a DB_DIRECTORY. This environment variable will specify a new directory on /var/opt/mssql/host_data/ within the container so that the DB may persist when adding other docker projects on the base image of mssql/server:2019-latest.
    
3. The ability to restore pre-existing databases from .mdf & .ldf files on `docker compose up`
    - If the container and its network are deleted, then the databases will be dropped. So when `docker compose up` runs again to create the containers, then the plain `CREATE DATABASE` command won't work since there is already a .mdf & .ldf file copied over. This commit 3ca4fc27fff44e0089b24a5fbd7ebd740ed462e1 will check if the .mdf file exists. If so, it will add a `FOR ATTACH` command to the end of `CREATE DATABASE` to restore the DB from .mdf and .ldf.

### CODE REVIEW

**Comment by JohanPetersson** [comment]
Thank you for this contribution. Before we can merge this, we need make this work a bit better with the templating system;

- For each template, exclude the `.env` if Docker isn't enabled, in these arrays https://github.com/episerver/content-templates/blob/main/templates/Alloy.Mvc/.template.config/template.json#L19.
- In each template's `template.json` there is a `sourceName`, this token gets replaced with the project's name when the template is applied. Use this token as DB name and directory in each template's `.env` file. E.g.:
```
SA_PASSWORD=Qwerty12345!
DB_NAME=Alloy.Mvc.1
DB_DIRECTORY=Alloy.Mvc.1
```

**Comment by gabepetersen** [comment]
@JohanPetersson I apologize for my late response to this. The recent commit [182b656](https://github.com/episerver/content-templates/pull/47/commits/182b656d6b8147586cf53f6ebc96d0441776c2eb) should update the code according to your recent comment. For the empty commerce template `.env` file, I added this to distinguish the two databases:
```
...
DB_NAME=Commerce.Empty.1_cms
DB_NAME_COMMERCE=Commerce.Empty.1_commerce
...
```

File: templates/Commerce.Empty/.env [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# These variables are used for passing values into docker-compose.yml and the containers
+SA_PASSWORD=Qwerty12345!
+DB_NAME=Commerce.Empty.1_cms
+DB_NAME_COMMERCE=Commerce.Empty.1_commerce
+DB_DIRECTORY=commerce-empty
```
Comment (JohanPetersson) [inline templates/Commerce.Empty/.env:5]:
```suggestion
DB_DIRECTORY=Commerce.Empty.1
```

File: templates/Alloy.Mvc/Docker/create-db.sh [hunk 2]
```diff
@@ -1,12 +1,23 @@
...
+    if [ $result -eq 0 ]; then
         echo "Creating database completed"
         break
     else
         echo "Creating database. Not ready yet..."
         sleep 1
     fi
-done
\ No newline at end of file
+done
```
Comment (JohanPetersson) [inline templates/Alloy.Mvc/Docker/create-db.sh:23]: I think this needs to be fixed on all three bash files... remove the empty last line. I know this empty line-ending can trip up on some systems. That's why we've added following in .editorConfig:

```
# Bash Files
[*.sh]
end_of_line = lf
insert_final_newline = false
```

**Review by JohanPetersson — CHANGES_REQUESTED** [review]
I've tested all templates now. Everything looks fine when the proposed changes I added now are in place. Thank you so much!

File: templates/Commerce.Empty/.env (hunk 1 above)
Comment (gabepetersen) [reply templates/Commerce.Empty/.env]: @JohanPetersson Fixed in [5d0b43e](https://github.com/episerver/content-templates/pull/47/commits/5d0b43ee2a5dcd7bccd2e4e3ffb0a5398688f3fd)

File: templates/Alloy.Mvc/Docker/create-db.sh (hunk 2 above)
Comment (gabepetersen) [reply templates/Alloy.Mvc/Docker/create-db.sh:23]: @JohanPetersson Fixed in [5d0b43e](https://github.com/episerver/content-templates/pull/47/commits/5d0b43ee2a5dcd7bccd2e4e3ffb0a5398688f3fd)

**Comment by gabepetersen** [comment]
@JohanPetersson Sorry again for the late response, I was out on vacation. Fixes to the recent change requests are in [4a9d57c](https://github.com/episerver/content-templates/pull/47/commits/4a9d57cc7a81435fe44ffbd71202ba9cafd14c67). Since I accidentally removed the newlines from the `.env` files instead of the `.sh` files in a commit I added, I had to force push a replace commit.

**Review by JohanPetersson — APPROVED** [review]
Thank you so much for your contribution again!

**Review by JohanPetersson — APPROVED** [review]

**Pull request merged**